glue/utils/matplotlib.py

```diff
@@ -176,3 +176,44 @@ def point_contour(x, y, data):
         return None
     xy = xy[0]
     return xy
+
+
+class AxesResizer(object):
+
+    def __init__(self, ax, margins):
+
+        self.ax = ax
+        self.margins = margins
+
+    def on_resize(self, event):
+
+        fig_width = self.ax.figure.get_figwidth()
+        fig_height = self.ax.figure.get_figheight()
+
+        x0 = self.margins[0] / fig_width
+        x1 = 1 - self.margins[1] / fig_width
+        y0 = self.margins[2] / fig_height
+        y1 = 1 - self.margins[3] / fig_height
+
+        dx = max(0.01, x1 - x0)
+        dy = max(0.01, y1 - y0)
+
+        self.ax.set_position([x0, y0, dx, dy]) 
+        self.ax.figure.canvas.draw()
+
+
+def fix_margins(axes, margins=[1, 1, 1, 1]):
+    """
+    Make sure margins of axes stay fixed.
+
+    Parameters
+    ----------
+    ax_class : matplotlib.axes.Axes
+        The axes class for which to fix the margins
+    margins : iterable
+        The margins, in inches. The order of the margins is
+        ``[left, right, bottom, top]``
+    """
+
+    resizer = AxesResizer(axes, margins)
```

Comment:
I don't think we should worry about future name clashes with resizer

On Friday, September 11, 2015, Thomas Robitaille <notifications@github.com>
wrote:

> In glue/utils/matplotlib.py
> <https://github.com/glue-viz/glue/pull/745#discussion_r39279062>:
>
> > +
> > +
> > +def fix_margins(axes, margins=[1, 1, 1, 1]):
> > +    """
> > +    Make sure margins of axes stay fixed.
> > +
> > +    Parameters
> > +    ----------
> > +    ax_class : matplotlib.axes.Axes
> > +        The axes class for which to fix the margins
> > +    margins : iterable
> > +        The margins, in inches. The order of the margins is
> > +        ``[left, right, bottom, top]``
> > +    """
> > +
> > +    resizer = AxesResizer(axes, margins)
>
> Hmm, that might be a way to customize the margins then, if it's accessible
> as axes.resizer.margins. I worry a bit about adding axes.resizer though
> in case it conflicts with Matplotlib in future for any reason. Do you think
> we should just do it anyway and cross that bridge if it happens?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/glue-viz/glue/pull/745/files#r39279062>.
>